### PR TITLE
Reduce redundancy between namedarray and variable tests

### PR DESCRIPTION
**Pull request by maxrjones**
Proof of concept for namedarray test structure proposed in #8370

### CODE REVIEW

File: xarray/tests/test_namedarray.py [hunk 1]
```diff
@@ -57,6 +57,35 @@ def __array_namespace__(self) -> ModuleType:
...
+    @pytest.fixture
+    def target(self):
+        """Fixture that needs to be re-declared"""
+        assert 0
+
+    @pytest.fixture
+    def data(self):
+        return 0.5 * np.arange(10).reshape(2, 5)
+
+    def test_properties(self, target, data):
```
Comment (dcherian) [inline xarray/tests/test_namedarray.py:70]:
```suggestion
    def test_properties(self, target, data) -> None:
```

File: xarray/tests/test_namedarray.py [hunk 2]
```diff
@@ -57,6 +57,35 @@ def __array_namespace__(self) -> ModuleType:
         return np
 
 
+class NamedArraySubclassobjects:
+    @pytest.fixture
+    def target(self):
+        """Fixture that needs to be re-declared"""
+        assert 0
```
Comment (max-sixty) [inline xarray/tests/test_namedarray.py:64]:
```suggestion
        raise NotImplementedError
```

File: xarray/tests/test_namedarray.py [hunk 3]
```diff
@@ -57,6 +57,35 @@ def __array_namespace__(self) -> ModuleType:
         return np
 
 
+class NamedArraySubclassobjects:
+    @pytest.fixture
+    def target(self):
+        """Fixture that needs to be re-declared"""
```
Comment (max-sixty) [inline xarray/tests/test_namedarray.py:63]: (nit)

```suggestion
        """Fixture that needs to be overridden"""
```

**Review by andersy005 — APPROVED** [review]
Looks solid to me 👍🏽! thank you, @maxrjones

**Pull request merged**